components/raftstore/src/store/fsm/apply.rs

```diff
@@ -959,15 +974,22 @@ where
         if !data.is_empty() {
             let cmd = util::parse_data_at(data, index, &self.tag);
 
+            let has_unsynced_data =
+                self.last_sync_apply_index < self.apply_state.get_applied_index();
+
             if apply_ctx.yield_high_latency_operation && has_high_latency_operation(&cmd) {
                 self.priority = Priority::Low;
                 if apply_ctx.priority != Priority::Low {
-                    apply_ctx.commit(self);
+                    if has_unsynced_data {
+                        apply_ctx.commit(self);
+                    }
                     return ApplyResult::Yield;
                 }
             }
             if should_write_to_engine(&cmd) || apply_ctx.kv_wb().should_write_to_engine() {
-                apply_ctx.commit(self);
+                if has_unsynced_data {
```

Comment:
If there is some admin request or write batch is too large, we must commit it....
I suggest you better handle these two situations(admin request and high latency operation) separately.